pool.go

```diff
@@ -567,7 +571,21 @@ func (p *Pool) Do(a Action) error {
 	}
 	defer p.put(c)
 
-	return c.Do(a)
+	startTime := time.Now()
+	err = c.Do(a)
+	p.traceDoCompleted(time.Since(startTime), err)
```

Comment:
The way this is implemented right now, the trace does not include the `get` or the `put` calls, and since it happens before the `put` it also won't ever show the Pool as being "full". I think `startTime` should initialize before `get`, and `put` should be taken out of the defer statement so `traceDoCompleted` can happen after it.

Reply:
I changed to initialize `startTime` at the first line of the `Do` function and call `traceDoComplete` right after `put` connection in `Pool`.

Reply:
Oh, I think I'm misunderstood. Did you mean to add a trace for `get`, `put` and also `DoComplete` events? How about making `PoolChange` or some trace for detecting `get`, `put` event in `Pool` and separate `DoComplete` trace?

Reply:
What you did here is what I had meant, no worries. I don't think it'd be useful to have `get` and `put` traces right now, I think `DoCompleted` is fine.

Reply:
@mediocregopher Sounds good, Thanks!